### PR TITLE
Remove `BelowSubsistenceThreshold` and `NewContractNotFunded`

### PR DESCRIPTION
**Pull request by cmichi**
Do not merge yet!

Closes https://github.com/paritytech/ink/issues/1044.

Follow-up to https://github.com/paritytech/substrate/pull/10082.

I haven't adapted the off-chain testing environment for the deposit mechanism, I'll create a follow-up for that. Right now, the off-chain env doesn't take the new deposit mechanism into account in any way.

### CODE REVIEW

**Review by athei — CHANGES_REQUESTED** [review]
Return codes didn't change for backwards compatibility. We merely stopped emitting some of them. Please check: https://github.com/paritytech/substrate/blob/450cd6804dd77312b27d72ebdebccc4c6027f579/frame/contracts/src/wasm/runtime.rs#L47-L77

**Comment by cmichi** [comment]
@athei Argh, ofc, I've updated the PR.

**Review by athei — APPROVED** [review]

File: examples/contract-transfer/lib.rs [hunk 1]
```diff
@@ -39,8 +24,8 @@ pub mod give_me {
         /// # Errors
         ///
         /// - Panics in case the requested transfer exceeds the contract balance.
-        /// - Panics in case the requested transfer would have brought the
-        ///   contract balance below the subsistence threshold.
+        /// - Panics in case the requested transfer would have brought this
+        ///   contract's balance below the minimum balance.
```
Comment (HCastano) [inline examples/contract-transfer/lib.rs:28]: What balance is the `minimum balance` referring to in this case?

**Review by HCastano — COMMENTED** [review]
What's this PR waiting for?

File: examples/contract-transfer/lib.rs (hunk 1 above)
Comment (cmichi) [reply examples/contract-transfer/lib.rs]: The chains existential deposit, I clarified it in the comment.

**Pull request merged**